tools/testoperator/src/telemetry.rs

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::sync::{Arc, LazyLock, OnceLock};
+
+use opentelemetry::metrics::{Gauge, Meter, MeterProvider};
+
+use opentelemetry_sdk::{Resource, metrics::SdkMeterProvider};
+use telemetry::noop::NoopMeterProvider;
+
+pub(crate) static METER_PROVIDER_ONCE: OnceLock<Arc<dyn MeterProvider + Send + Sync>> =
```

Comment:
I would put this code in the `test-framework` crate, not in the `testoperator` binary directly. That way we could reuse it for other more specialized binaries in the future if we needed to.

Reply:
I would also split the meter/exporter creation from the individiual metrics themselves.